staging/src/k8s.io/kubectl/pkg/cmd/wait/wait.go

```diff
@@ -152,6 +156,7 @@ func (flags *WaitFlags) AddFlags(cmd *cobra.Command) {
 
 	cmd.Flags().DurationVar(&flags.Timeout, "timeout", flags.Timeout, "The length of time to wait before giving up.  Zero means check once and don't wait, negative means wait for a week.")
 	cmd.Flags().StringVar(&flags.ForCondition, "for", flags.ForCondition, "The condition to wait on: [delete|condition=condition-name[=condition-value]|jsonpath='{JSONPath expression}'=[JSONPath value]]. The default condition-value is true.  Condition values are compared after Unicode simple case folding, which is a more general form of case-insensitivity.")
+	cmd.Flags().BoolVar(&flags.WaitForCreation, "wait-for-creation", flags.WaitForCreation, "If set to true, also wait for creation of objects if they do not already exist. This flag can not be used with --for=delete or --timeout=0")
```

Comment:
Why entirely new flag, rather than building on the existing `--for=create`?

Reply:
I think you need things in combination, e.g. wait for create and then for a condition, so that makes it a little non-obvious how to accomplish that.

You could allow stacking `--for`, like this:

```
--for=create --for=jsonpath=...
```

Reply:
This alternative was discussed in https://github.com/kubernetes/kubernetes/pull/122994#discussion_r1497871808 and concluded that new flag will make users life easier (one command execution vs two).

Reply:
> I think you need things in combination, e.g. wait for create and then for a condition, so that makes it a little non-obvious how to accomplish that.
> 
> You could allow stacking `--for`, like this:
> 
> ```
> --for=create --for=jsonpath=...
> ```

yes, that was the major reason actually.

Reply:
I'd probably go with stacking, rather than adding new flags for each case. If soon enough we also decide to expand this to support waiting for an update, would we add `--wait-for-update`, or better have `--for=update`. I'm personally leaning towards expanding the functionality of `--for` rather than adding more flags. We keep rejecting PRs adding more flags, saying it's harder to maintain additional flags, but we don't seem to keep ourselves to that premise.

Reply:
People would require this only for creation (not for update or delete) and there is even a demand to make this flag's default as true. `--for` will duplicate the code and won't be different then current recommended workaround (wait creation with your script or with our `--for=creation` and execute your actual wait).
Especially this comment https://github.com/kubernetes/kubernetes/pull/122994#discussion_r1617946169 explains better.

Reply:
I think, if we go with flag route, at some point we can switch it to true as default and people intuitively run wait command assuming that creation will also be waited. OTOH, if we go with `--for`, people will be forced to run 2 subsequent commands.

Reply:
I am really unsure about defaulting the "wait for create" behavior to true. It's a useful thing to be able to opt into, for sure, but switching cube control to hang by default if the object you're waiting on some aspect of doesn't exist is not what I would expect.

Reply:
Are you suggesting something like this;

```sh
kubectl wait --for=create --for=jsonpath='{.status.phase}'=Running pod/busybox1
```

Additionally, @liggitt also mentioned above never changing the default to true. This ^ makes sense, although it requires refactoring in the kubectl wait.